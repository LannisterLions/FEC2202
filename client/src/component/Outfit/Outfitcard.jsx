import React, { useState, useEffect } from "react";
import OutfitCSS from "../cssModules/Outfit.module.css";
import PropTypes from "prop-types";
import axios from "axios";
import { starReview } from "../functions.jsx";

function Outfitcard(props) {
  const [averagescore, setAveragescore] = useState(0);
  //const [starreview, setStarreivew]=useState(<div></div>)

  useEffect(() => {
    axios
      .get("/reviews/meta", { params: { product_id: props.item.id } })
      .then((res) => {
        let average = 0;
        if (!res.data.ratings) {
          setAveragescore(0);
        } else {
          let counter = 0;
          let sumscore = 0;
          for (const [star, num] of Object.entries(res.data.ratings)) {
            counter += Number(num);
            sumscore += Number(star) * Number(num);
          }
          //console.log("counter,sumscore",counter,sumscore)
          average = sumscore / counter;
          setAveragescore(average);
        }
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);
  //console.log("averagescore>>>>",averagescore)
  const handledelete = () => {
    let collection = JSON.parse(localStorage.getItem("collection"));
    // console.log("getting inforamtion from localstorage>>>>",collection)
    // console.log(props.item.id)
    let newcollection = collection.filter(
      (collectitem) => props.item.id !== collectitem.id
    );
    //console.log("after filter collection",newcollection)
    localStorage.setItem("collection", JSON.stringify(newcollection));
    props.handleUpdate();
  };

  return (
    <div className={OutfitCSS.card} enzyme-test="outfitcard">
      <div className={OutfitCSS.imageparent}>
        <button
          className={OutfitCSS.outfitcardbutton}
          onClick={handledelete}
          enzyme-test="deleteoutfitbutton"
        >
          <i className="fa fa-times fa-2x" aria-hidden="true"></i>
        </button>
        <img className={OutfitCSS.image} src={props.item.image}></img>
      </div>
      <p>{props.item.category}</p>
      <h3>{props.item.name}</h3>
      {starReview(averagescore, OutfitCSS)}

      {/* <h2>review star placeholder</h2> */}
      {/* <div style={{"background-color":"yellow"}}>
      {<div className={OutfitCSS.Stars} style={{"--rating": 2.3}} aria-label={"Rating of this product is 2.3 out of 5."}></div> }
      <div style={{"backgroud":"linear-gradient(90deg, #eee 30%, #fff 70%"}}>☆☆☆☆☆ </div>  
      </div>*/}
      <span>
        <strong>${props.item.default_price}</strong>
      </span>
    </div>
  );
}

Outfitcard.propTypes = {
  item: PropTypes.object,
  handleUpdate: PropTypes.func,
};

export default Outfitcard;
