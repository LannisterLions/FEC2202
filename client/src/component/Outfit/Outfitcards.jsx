
// class Outfitcards extends React.Component {
//   constructor(props) {
//     super(props);
//     this.state = {};
//   }

//   render() {
//     return (
//       <div className="outfitcards">
//         <h1>Outfitcards</h1>
//       </div>
//     );
//   }
// }

let fakelocalstoragedata=[{
  "id": 65637,
    "name": "Blues Suede Shoes",
    "category": "Dress Shoes",
    "default_price": "120.00",
    "image":"https://images.unsplash.com/photo-1561861422-a549073e547a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
},
{"id": 65644,
"name": "Greg Sweatpants",
"category": "Sweatpants",
"default_price": "599.00",
"image":"https://images.unsplash.com/photo-1555274175-6cbf6f3b137b?ixlib=rb-1.2.1&auto=format&fit=crop&w=300&q=80"
}
]

import React,{ useState, useEffect } from 'react'
import OutfitCSS from '../cssModules/Outfit.module.css'
import Outfitcard from './Outfitcard.jsx'
import PropTypes from 'prop-types'

function Outfitcards(props) {
const [collection,setCollection]=useState(()=>{
  let saved=localStorage.getItem("collection");
  let initialValue=JSON.parse(saved);
  return initialValue || [];
})

console.log(localStorage)
//populate the state from local storage
//onclick, collect the prop data(item name, price, review )
//to save it to the storage

let handleAdd = ()=>{
  let newAdd = {}
  newAdd.id=props.currProductData.id;
  let collection=localStorage.getItem("collection");
  if (collection.indexOf(`"id":${newAdd.id}`)===-1) {
    //console.log("collectionindexof>>>>",collection.indexOf(`"id":${newAdd.id}`))
  newAdd.name=props.currProductData.name;
  newAdd.category=props.currProductData.category
  newAdd.default_price=props.currStyle.original_price;
  newAdd.image=props.currStyle.photos[0].thumbnail_url || "https://whetstonefire.org/wp-content/uploads/2020/06/image-not-available.jpg"
  collection=JSON.parse(collection)
  collection.push(newAdd)
  localStorage.setItem("collection",JSON.stringify(collection))
  setCollection(collection)
  return;
  } 
}

let handleUpdate =()=>{
  let collection=JSON.parse(localStorage.getItem("collection"));
  setCollection(collection)
}
useEffect(()=>{
  
let collection=JSON.parse(localStorage.getItem("collection"))
  // console.log("localStorage>>>>",localStorage)
  // console.log("collection>>>>",collection)
  setCollection(collection)
},[])


  return (
    <>
    <h1>My Collection</h1>
    <div className={OutfitCSS.container}>
    <div className={OutfitCSS.card}>
      <h1 className={OutfitCSS.text} onClick={handleAdd}>Add this outfit to your collection</h1>
      </div>
    { collection? collection.map((item)=><Outfitcard key={item.id} item={item} handleUpdate={handleUpdate}/>) : ""
    }
    </div>
    </>
  )
}

Outfitcards.propTypes = {
  currStyle: PropTypes.object,
  currentProduct:PropTypes.number,
  currProductData:PropTypes.object

 
  
};


export default Outfitcards