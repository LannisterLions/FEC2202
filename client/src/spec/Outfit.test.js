import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json';
configure({ adapter: new Adapter() });
import Outfitcard from '../component/Outfit/Outfitcard.jsx';
import Outfitcards from '../component/Outfit/Outfitcards.jsx';


describe ("Outfit card test",()=>{

    let fakedata= {
        category:"test",
        default_price: "123",
        id: 123,
        image:"http://dasacounseling.weebly.com/uploads/5/6/1/4/56149545/2263864_orig.jpg",
        name: "testing product"
    }

    it('Renders outfit card ',()=>{
        shallow(< Outfitcard item={fakedata}/>)
    })
    
    test("match snapshot", ()=>{
        let wrapper = shallow(<Outfitcard item={fakedata}/>);
        expect(toJSON(wrapper)).toMatchSnapshot()
    })


})