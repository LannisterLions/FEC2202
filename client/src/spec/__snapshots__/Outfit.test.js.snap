// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outfit card test match snapshot 1`] = `
<div
  className="card"
>
  <div
    className="imageparent"
  >
    <button
      className="outfitcardbutton"
      onClick={[Function]}
    >
       
      <img
        src="./cross.png"
      />
    </button>
    <img
      className="image"
      src="http://dasacounseling.weebly.com/uploads/5/6/1/4/56149545/2263864_orig.jpg"
    />
  </div>
  <h3>
    testing product
  </h3>
  <p>
    test
  </p>
  <h2>
    review star placeholder
  </h2>
  <h3>
    123
  </h3>
</div>
`;
